HIP/hip-792.md

```diff
@@ -0,0 +1,290 @@
+
+## Motivation
+
+<!-- The motivation is critical for HIPs that want to change the Hedera codebase or ecosystem. It should clearly explain why the existing specification is inadequate to address the problem that the HIP solves. HIP submissions without sufficient motivation may be rejected outright. -->
+
+`hederaAccountService.isAuthorized(address, messageHash, signatureBlob)` does work for the following scenarios:
+
+- When an account's admin key is "simple", and is comprised of either a single EdDSA key, or a single ECDSA key
+- When an account's admin key is "complex", and is comprised of multiple EdDSA keys and/or  ECDSA keys, combined using one or more `KeyList`s or `ThresholdKey`s.
+
+`hederaAccountService.isAuthorized(address, messageHash, signatureBlob)` does **not** work for the following scenarios:
+
+- When an account's admin key is "complex", and is comprised of multiple EdDSA keys and/or  ECDSA keys and/or **smart contract IDs**, combined using one or more `KeyList`s or `ThresholdKey`s.
+
+The proposed `hederaAccountService.isAuthorizedCurrentTransaction()` method aims to fulfil the above scenario, and specific use cases pertaining to this scenario will be elaborated upon in the [use cases section](#use-cases) below.
+
+Furthermore, Hedera Token Service exposes an authorization mechanism already,
+which is capable of handling this scenario that is presently unfulfilled by Hedera Account Service as described in HIP-632.
+<!-- TODO links/ references that demonstrate this in HTS -->
+
+> NOTE:
+>
+> - "EdDSA" refers to **EdDSA ED25519**, and
+> - "ECDSA" refers to **ECDSA secp256k1** throughout this HIP, unless otherwise stated.
+>
+> These are the only types of cryptographic keys currently supported by Hedera's account system.
+
+## Rationale
+
+<!-- The rationale fleshes out the specification by describing why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g., how the feature is supported in other languages. The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during the discussion. -->
+
+In existing code examples where `ecrecover` is used,
+a smart contract function is passed in the transaction data in its parameters.
+The design of both `hederaAcountService.isAuthorized(address, messageHash, signatureBlob)`
+and `hederaAcountService.isAuthorizedRaw(address, messageHash, signatureBlob)`
+in HIP-632 mimics this approach.
+
+This approach is adequate in situations where we assume that:
+
+(1) All authorization is performed using cryptographic signatures (EdDSA and/or ECDSA in the case of Hedera).
+
+(2) All authorization is performed "off-chain", that is among clients, and not within (or by) smart contracts.
```

Comment:
Okay you should say signing then. Saying authorization is done off-chain might give the wrong impression

Reply:
Hmm, the reason I use the word authorisation over signing is because the ECDSA and EdDSA keys can sign a transaction, but a smart contract ID does not. So I'm using a term that is inclusive of both "cryptographic signature based authorisation" and "smart contract invocation based authorisation".

Is there perhaps another term (other than "authorisation") which covers both of these?